compiler/rustc_target/src/spec/targets/i686_unknown_uefi.rs

```diff
@@ -86,7 +32,7 @@ pub(crate) fn target() -> Target {
         },
         pointer_width: 32,
         data_layout: "e-m:x-p:32:32-p270:32:32-p271:32:32-p272:64:64-\
-            i64:64-i128:128-f80:32-n8:16:32-a:0:32-S32"
+            i128:128-f64:32:64-f80:32-n8:16:32-S128"
```

Comment:
Running `./x test tests/assembly/targets/targets-pe.rs` shows that [the data layout](https://llvm.org/docs/LangRef.html#data-layout) for our `i686-unknown-uefi`, for it to match Clang's `i686-unknown-uefi`, should have the following modifications:
- Remove `i64:64`: Decreases the alignment of `i64` to 32 bits.
- Add `f64:32:64`: Decreases the alignment of `f64` to 32 bits.
- Remove `a:0:32`: Decreases the alignment of aggregates (structs?) to 32 bits.
- Replace `S32` with `S128`: Increases the natural alignment of the stack to 128 bits.

I have no idea if this is the desired behaviour?

Reply:
> Remove i64:64: Decreases the alignment of i64 to 32 bits.
> Add f64:32:64: Decreases the alignment of f64 to 32 bits.

These I think are not desirable. Edk2 sets [`-malign-double`](https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html#index-malign-double) for ia32 in https://github.com/tianocore/edk2/blob/HEAD/BaseTools/Conf/tools_def.template. So while the UEFI spec isn't super clear on what the alignment for `double` and `long long` should be on ia32, the defacto alignment is 8.

See https://github.com/rhboot/shim/pull/516 for a real-world example where this makes a difference.

Reply:
can we file a bug against LLVM, then?

Reply:
I would prefer if @nicholasbishop did it, you seem more knowledgeable about it - but I can try to do it if you don't have the time?

Reply:
I just did a quick sanity check of compiling C code with these triples, and I suspect this isn't fully implemented in clang yet (tested 19.1.0). Getting errors like: `error: backend data layout 'e-m:w-p270:32:32-p271:32:32-p272:64:64-i64:64-i128:128-f80:128-n8:16:32:64-S128' does not match expected target description 'e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-i128:128-f80:128-n8:16:32:64-S128'`.

I think the implementation work in LLVM is ongoing, so not quite ready for us to use yet. @Prabhuk can you confirm that?

Reply:
Breaking from edk2 is definitely not desirable. From what I understand, even most of the proprietary UEFI implementations use edk2 as the base.

Reply:
Apologies for the slow response! I am on vacation till mid November. I have a WIP backend patch (https://github.com/llvm/llvm-project/pull/109320)  to support X86_64 first. I am debugging a assertion failure I ran into while compiling a downstream EFI example. I will prioritize upstreaming fixing it and sending out a PR as soon as I return. aarch64 will be next. w.r.t to data layout, I need to look at a closer look once I return before commenting.

Reply:
CC: @frobtech @petrhosek